Draft: Action server docs plugins (#12333)
RasaHQ/rasa: docs/docs/action-server/add-sanic-extensions.mdx

---
id: add-sanic-extensions
sidebar_label: Adding Sanic Extensions
title: Add Sanic Extensions
---

# Add Sanic extensions

> With the use of Pluggy, you can now create additional sanic extensions by accessing the app object created by the action server. The hook implemented provides you access 

Reviewer: This is an implementation detail, we can leave this out:
```suggestion
You can now create additional Sanic extensions by accessing the app object created by the action server. The hook implemented provides you access 
```

Reviewer: Could you please address this first, it's removing the implementation reference to pluggy.

> to sanic app object created by the rasa sdk when starting the action server. Using this app object, you can now extend Sanic features such as middlewares, listeners, background tasks

Reviewer: ```suggestion
to the Sanic app object created by `rasa-sdk` when starting the action server. Using this app object, you can now extend Sanic features such as middlewares, listeners, background tasks
```

Reviewer: This should be addressed too.

Author: i think my rebase overwrote all the changes i did here :( i will redo it.

and additional routes.

### Step by Step Guide on creating your own sanic extension in rasa_sdk
This example will show how to create a sanic listener using plugins.

#### Create package rasa_sdk_plugins
Create a package in your action server project called `rasa_sdk_plugins`. Rasa SDK will try to instantiate this package in your project to start plugins.

> If no plugins found, it will print an info that there are no plugins in your project.

Reviewer: ```suggestion
If no plugins are found, it will print an info log that there are no plugins in your project.
```

#### Register modules containing the hooks

> Instantiate the package `rasa_sdk_plugins` and initialise the hooks. create an `__init__.py` Plugin manager will look for the module where the hooks are implemented

Reviewer: ```suggestion
Create the package `rasa_sdk_plugins` and initialize the hooks by creating an `__init__.py` file where the plugin manager will look for the module where the hooks are implemented:
```
